discovery/azure/azure.go

```diff
@@ -508,3 +509,17 @@ func (client *azureClient) getNetworkInterfaceByID(networkInterfaceID string) (n
 
 	return result, nil
 }
+
+func getPowerStatusFromVMInstanceView(instanceView *compute.VirtualMachineInstanceView) string {
```

Comment:
It could rewritten a bit for a better [line of sight](https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88):

```go
func getPowerStatusFromVMInstanceView(instanceView *compute.VirtualMachineInstanceView) string {
    if instanceView.Statuses == nil {
        return ""
    }
    var powerStatusCode string
    for _, ivs := range *instanceView.Statuses {
        if strings.HasPrefix(*ivs.Code, "PowerState") {
            powerStatusCode = *ivs.Code
        }
    }
    return powerStatusCode
}
```